flytekit/types/structured/structured_dataset.py

```diff
@@ -820,13 +820,36 @@ def _get_dataset_column_literal_type(self, t: Type) -> type_models.LiteralType:
             return type_models.LiteralType(map_value_type=self._get_dataset_column_literal_type(t.__args__[1]))
         raise AssertionError(f"type {t} is currently not supported by StructuredDataset")
 
+    def flatten_dict(self, nested_dict):
+        result = {}
+
+        def _flatten(sub_dict, parent_key=""):
```

Comment:
Why do we need another function here? can we do 

```
def flatten_dict(self, nested_dict):
    result = {}
    for key, value in sub_dict.items():
        ...
    return result
```

Reply:
I think have to do it recursively, since it's nested for a undetermined depth. Or can we do it iteratively?